build_tools/build_detect_platform

```diff
@@ -134,7 +134,7 @@ case "$TARGET_OS" in
         ;;
     FreeBSD)
         PLATFORM=OS_FREEBSD
-        COMMON_FLAGS="$COMMON_FLAGS -fno-builtin-memcmp -D_REENTRANT -DOS_FREEBSD"
+        COMMON_FLAGS="$COMMON_FLAGS -fno-builtin-memcmp -D_REENTRANT -DOS_FREEBSD -D_GLIBCXX_USE_C99 -fpermissive"
```

Comment:
@avsej Okay cool. Let me bring up a FreeBSD VM to test this on, and all being well I will merge.